### PR TITLE
session: enable `foreign_key_checks`/`tidb_enable_foreign_key` by default when upgrade

### LINKED ISSUE #41423: system variables which default value changed should use new default value after upgrade
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
if use not config these system variable, default value should be new value after upgrade.
foreign_key_checks= old:'OFF' new: 'ON'
tidb_enable_foreign_key= old:'OFF' new: 'ON'
tidb_enable_historical_stats= old:'OFF' new: 'ON'
tidb_enable_plan_replayer_capture= old:'OFF' new: 'ON'
tidb_store_batch_size= old:'0' new: '4'

### 2. What did you expect to see? (Required)
default value should be same with new scratch cluster
### 3. What did you see instead (Required)
still use old default value
### 4. What is your TiDB version? (Required)

v6.6.0-pre

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: close #41423

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/bootstrap.go [hunk 1]
```diff
@@ -2285,6 +2290,15 @@ func upgradeToVer112(s Session, ver int64) {
 	doReentrantDDL(s, CreateMDLView)
 }
 
+func upgradeToVer113(s Session, ver int64) {
+	if ver >= version113 {
+		return
+	}
+
+	mustExecute(s, "REPLACE HIGH_PRIORITY INTO %n.%n VALUES (%?, %?);", mysql.SystemDB, mysql.GlobalVariablesTable, variable.ForeignKeyChecks, variable.On)
```
Comment (you06) [inline session/bootstrap.go:2298]: Why use replace instead of insert ignore? It'll force update the previous value.

Comment (crazycs520) [reply session/bootstrap.go:2298]: `insert ignore` won't take any effect since the variable_name already exist.

**Review by Yisaer — APPROVED** [review]

File: session/bootstrap_test.go [hunk 2]
```diff
@@ -1417,3 +1417,70 @@ func TestTiDBGCAwareUpgradeFrom630To650(t *testing.T) {
...
+		require.Equal(t, 1, chk.NumRows())
+		row := chk.GetRow(0)
+		require.Equal(t, 2, row.Len())
+		require.Equal(t, varsValueList[i], row.GetString(1))
+	}
+}
+
+func TestTiDBGlobalVariablesDefaultValueUpgradeFrom650To660(t *testing.T) {
+
+}
```
Comment (crazycs520) [inline session/bootstrap_test.go:1486]:
```suggestion
```

**Review by you06 — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**